kernel/libs/aster-bigtcp/src/iface/poll.rs

```diff
@@ -509,4 +632,75 @@ impl<E: Ext> PollContext<'_, E> {
 
         (did_something, tx_token)
     }
+
+    fn dispatch_raw<T, Q>(&mut self, tx_token: T, dispatch_phy: &mut Q) -> (bool, Option<T>)
+    where
+        T: TxToken,
+        Q: FnMut(&Packet, &mut Context, T),
+    {
+        let mut tx_token = Some(tx_token);
+        let mut did_something = false;
+        for socket in self.sockets.raw_socket_iter() {
+            if !socket.need_dispatch(self.iface_cx.now()) {
+                continue;
+            }
+
+            // We set `did_something` even if no packets are actually generated. This is because a
+            // timer can expire, but no packets are actually generated.
+            did_something = true;
+
+            let mut deferred = None;
+
+            socket.dispatch(self.iface_cx, |cx, ip_repr, ip_payload| {
+                let this = PollContext::new(cx, self.sockets, self.new_tcp_conns);
+                if ip_repr.dst_addr().is_broadcast() || !this.is_unicast_local(ip_repr.dst_addr()) {
+                    dispatch_phy(
```

Comment:
If it's not a broadcast packet, we should make raw packets capture such packets. (Broadcast packets will be captured in `parse_and_process_ipv4` anyway).